### ISSUE TITLE
Spike: Use Renovate to update dependencies in non-standard files

### ISSUE THREAD

**Issue by czeslavo** [Kong/kubernetes-ingress-controller#4687]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem Statement

Try to use Renovate's [regex managers](https://docs.renovatebot.com/modules/manager/regex/) to automatically update dependencies defined in non-standard files (e.g. Kind docker image in GH workflows, Kong Helm chart in integration tests const, etc.). 

### Proposed Solution

Check if Renovate's regex managers are able to solve our issue of updating dependencies in non-standard files (not handled by currently used Dependabot).

### Acceptance Criteria

- [x] KIC fork exists with Renovate regex managers definitions that allow updating one of dependencies defined in non-standard files

**Comment by czeslavo** [comment]
I verified that Renovate is a good candidate for updating dependencies defined in non-standard files like GH workflows or any other custom format we might come up with in #4691. 

I made it work to update a Kong Helm chart version in go sources and docker image in a workflow file with no huge difficulties.

The configuration used is defined [here](https://github.com/czeslavo/kubernetes-ingress-controller/blob/1a7c17741adb1707a84554116b620e21fa4c7887/.github/renovate.json).
- It disables all the default managers that are already handled by Dependabot (go.mod, Dockerfile, etc.).
- It defines two regex managers: Kong Helm chart in `helm.go` and Kind docker image in `_e2e_tests.yaml`. 
- It disables minor bumps for Kind with the use of `packageRules`

Sample PRs generated by Renovate:
- https://github.com/czeslavo/kubernetes-ingress-controller/pull/8 
- https://github.com/czeslavo/kubernetes-ingress-controller/pull/6
